### DISCUSSION TITLE
Vercel detects 'eleventy' as my framework, how do I set  'other' as my frontend framework?

### DISCUSSION THREAD

**Discussion by dbauszus-glx** [vercel/vercel#5162] [Help]
I have eleventy installed as a dev dependency and use this locally to build my documentation pages. But I do not use a frontend framework and would like to chose 'other'. This seems to be only possible in the Vercel dashboard but not through the CLI / Vercel.json. Or maybe it is but not documented or I can not find this in the documentation.

Defining an empty build in the package.json does not seem to do the trick.

How can I get a single serverless function to be build with the node builder?

Do I need to define the runtime?

Is there some hidden build command for the node.js builder?

How can I force the frontend framework to be other bar uninstalling eleventy from the dev dependencies as well as removing all eleventy ressources from the project?

```
"functions": {
  "api/api.js": {
    "includeFiles": "public/**"
  }
}
```

**Comment by dbauszus-glx** [comment]
@paulogdm suggested to override the build command, output directory, and development command which would be required for an 'other' build. Unfortunately I am not able to do any selection in the CLI. 'Enter' will continue the deployment as is. Other keys do not have any effect.

```
dennis@dennis:~/Git/xyz$ vercel
Vercel CLI 20.1.0
? Set up and deploy “~/Git/xyz”? [Y/n] y
? Which scope do you want to deploy to? GEOLYTIX
? Link to existing project? [y/N] n
? What’s your project’s name? xyz-mapp-dev
? In which directory is your code located? ./
Auto-detected Project Settings (Eleventy):
- Build Command: `npm run build` or `npx @11ty/eleventy`
- Output Directory: _site
- Development Command: npx @11ty/eleventy --serve --watch --port $PORT
? Want to override the settings? [y/N] y
? Which settings would you like to overwrite (select multiple)? 
  ▪︎ Build Command
› ▪︎ Output Directory
  ▪︎ Development Command
```

I would like these settings to be the default for 'other' deployments.

```
- Build Command: `npm run vercel-build` or `npm run build`
- Output Directory: `public` if it exists, or `.`
- Development Command: None
```

**Reply by dbauszus-glx** [reply]
Is it possible to set defaults in the package.json? Even if this would work, it wouldn't be great having to overwrite the defaults to defaults for every deployment because a dev dependency is detected which should not even be installed in production deployments.

**Comment by styfle** [comment]
I believe the confusion here is that you need to use "Spacebar" instead of "Enter" when using the multichoice selector with the CLI. Enter is going to finalize your choices.

Alternatively, you can type "a" and then "Enter" to select all.

**Reply by dbauszus-glx** [reply]
@styfle Thanks for the suggestion. That does work to some extent. May I suggest though to add the information in regard to the keys to the CLI text? I would also suggest a better character set to indicate whether lines a selected. Please compare these screenshots.

![image](https://user-images.githubusercontent.com/22201617/92916519-8032c680-f425-11ea-969b-7179899fcb68.png)

I believe all lines are selected in this image.

![image](https://user-images.githubusercontent.com/22201617/92916585-8f197900-f425-11ea-8472-81466f533bb7.png)

More importantly though... How do I set None? Is it just leaving the line override empty?

Checking the Vercel checkboard still suggest Eleventy with the overrides active.

The build command does not work neither.

Deploying defaults with eleventy removed works but like this vercel-build script is missing. `npm ERR! missing script: vercel-build`

![image](https://user-images.githubusercontent.com/22201617/92918885-ba04cc80-f427-11ea-84a2-a13df87b24ae.png)

**Reply by styfle** [reply]
The error message is expected because eleventy doesn't ship with a `vercel-build` script, only the `build` script as seen here: https://github.com/vercel/vercel/blob/72572ba6be3e3de8a630bc0e9745406af4c4a565/examples/eleventy/package.json#L5-L9

Does your code provide a `vercel-build` script?

**Reply by dbauszus-glx** [reply]
Oh, I think there is a misunderstanding on my side. Isn't `npm run vercel-build` the default which is required for no frameworks?

My code doesn't provide vercel-build.

**Reply by styfle** [reply]
Your screenshot shows you that you manually entered a Build Command which will override the default.

**Reply by dbauszus-glx** [reply]
Yes, because the default was `eleventy` and my understanding was that the default for no framework is `vercel-build`.

**Reply by dbauszus-glx** [reply]
I have now uninstalled eleventy in project and use global install. I'd prefer if the autodetect doesn't go down this route. But I rather get going and not worry about eleventy as a project dependency.

**Reply by styfle** [reply]
There should not be a difference between Framework=Other vs Framework=Eleventy if you override all the other settings.

The purpose of Other is a placeholder to mean the framework could not be auto-detected.

**Comment by styfle** [comment]
It might be easier to use the git import web interface here: https://vercel.com/import/git
